### DISCUSSION TITLE
Q26 ("Biggest Pain Point") should be per-choice ratings

### DISCUSSION THREAD

**Discussion by peter-mlabs** [cardano-foundation/state-of-the-developer-ecosystem#3] [General feedback]
As presently stated, this question and the resulting conclusions seem to hide a lot of information. The results tell us, for instance, that 26 people found documentation to be the _most_ painful point of cardano; but all this tells us is that 114 of the respondents _did not_ find this the most painful. We have no indication of what those other 114 people felt about it; perhaps it was the #2 for 100 of them, perhaps it wasn't a concern at all for these 100 people.

As such, the conclusion seems that is can only be properly stated as

> The most prominent pain point **_among a minority of respondents_** in Cardano's developer ecosystem is the scattered and sometimes hard-to-find documentation, particularly for newcomers seeking entry-level guidance. There's a strong sentiment **_among a minority of respondents_** about the need for better-organised, more accessible, and distinguishable quality resources.

Like wise 

> The heavy reliance on Haskell is seen as a double-edged sword; while it underpins the network's robustness, it also poses a steep learning curve and potentially slows broader adoption due to its niche status. This perception extends to the use of Nix, adding another layer of complexity for those unfamiliar with these technologies.

A _minority of respondents_ (at least 6 of whom _were inexperienced in haskell according to the extended responses_)  reported haskell as their biggest pain point. Listing this as the "#2" pain-point or "a double edged-sword" is an extremely strong conclusion when ~80% of respondents _did not find this to be the **most** painful part of cardano_. If seen this question referenced as an indication that cardano needs to divest from Haskell, and I don't believe the data supports such a conclusion at all.

The remainder of this question and the inferred conclusions follow similarly. This question would give much more meaningful information if it asked respondents to rate how painful each technology is on, say, a scale from 1-5.

**Comment by KtorZ** [comment]
Keep in mind that this was **an open question**. So it isn't much that "n people amongst m found X hard", but rather that "n people amongst X thought by themselves that X was hard". This gives even more weight to points being mentioned by participants. When, on a completely open question, 14% of the respondents think about 'Haskell' as the answer they want to give to the most painful point, I find it rather insightful. That they are young and inexperienced doesn't invalidate their opinion. 

Also, you're perhaps over-interpreting the interpretation? I don't think anyone said we needed to divest from Haskell. But rather that it has a steep learning curve and that it is currently perceived as a pain point. Does it mean Haskell is bad? No. Does it mean we need more effort to make Haskell more accessible? Perhaps. At the end of the day, the strategic decision is up to the builder and the community. If the effort of "making Haskell better and more accessible" is larger than moving to another language ecosystem, then moving away from Haskell may be a more viable option. If we manage to make Haskell more approachable and usable overall, then it will cease to be a pain point.
